packages/neuron-ui/src/components/TransactionFeePanel/index.tsx

```diff
@@ -1,81 +1,143 @@
-import React, { useState } from 'react'
-import { Stack, Toggle, IDropdownOption } from 'office-ui-fabric-react'
-import Dropdown from 'widgets/Dropdown'
+import React, { useState, useMemo, useRef } from 'react'
+import clsx from 'clsx'
+import { Stack } from 'office-ui-fabric-react'
+import { useClickAway } from 'ahooks'
 import TextField from 'widgets/TextField'
 import { useTranslation } from 'react-i18next'
 import { Price, localNumberFormatter } from 'utils'
+import Button from 'widgets/Button'
+import { ArrowActive, ArrowClose, Select } from 'widgets/Icons/icon'
+
+import styles from './transactionFeePanel.module.scss'
 
 interface TransactionFeeProps {
   fee: string
   price: string
   onPriceChange: any
 }
 
+interface PriceObj {
+  key: string
+  text: string[]
+  color: string
+}
+
+interface SelectItemProps {
+  priceObj: PriceObj
+  className?: any
+  onClick: any
+  sufIcon?: any
+}
+
+const priceOptions: PriceObj[] = [
+  { key: Price.Low, text: ['send.slow', Price.Low], color: 'blue' },
+  { key: Price.Medium, text: ['send.standard', Price.Medium], color: 'green' },
+  { key: Price.High, text: ['send.fast', Price.High], color: 'red' },
+]
+
+const SelectItem: React.FunctionComponent<SelectItemProps> = ({
+  priceObj = priceOptions[0],
+  className,
+  sufIcon,
+  ...res
+}: SelectItemProps) => {
+  const [t] = useTranslation()
+  return (
+    <Button type="text" className={clsx(className, styles['select-item'])} {...res}>
+      <div className={styles.wrap}>
+        <div>
+          <p className={styles.title}>{t(priceObj.text[0])}</p>
+          <p className={clsx(styles[priceObj.color], styles.tag)}>
+            {t('send.price')} {priceObj.text[1]} shannons/kB
+          </p>
+        </div>
+        <p>{sufIcon}</p>
+      </div>
+    </Button>
+  )
+}
+
 const TransactionFee: React.FunctionComponent<TransactionFeeProps> = ({
   price,
   fee,
   onPriceChange,
 }: TransactionFeeProps) => {
   const [t] = useTranslation()
-  const [showDetail, setShowDetail] = useState(false)
+  const [isCustom, setIsCustom] = useState(false)
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false)
+  const dropdownRef = useRef<HTMLDivElement>(null)
+
+  useClickAway(() => {
+    if (isDropdownOpen) {
+      setIsDropdownOpen(false)
+    }
+  }, dropdownRef)
+
+  const selectedPrice = useMemo(() => priceOptions.filter(item => item.key === price)?.[0], [price])
```

Comment:
`Array#.find` could be used here

Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find